doc/reference/configuration/configuration_reference.rst

```diff
@@ -2389,3 +2389,345 @@ The ``security`` section defines configuration parameters related to various sec
+
+..  _configuration_reference_wal:
+
+wal
+---
+
+The ``wal`` section defines configuration parameters related to :ref:`write-ahead log <internals-wal>`.
+To learn more about the WAL configuration, check the :ref:`Persistence <configuration_persistence_wal>` page.
+
+..  NOTE::
+
+    ``wal`` can be defined in any :ref:`scope <configuration_scopes>`.
+
+-   :ref:`wal.cleanup_delay <configuration_reference_wal_cleanup_delay>`
+-   :ref:`wal.dir <configuration_reference_wal_dir>`
+-   :ref:`wal.dir_rescan_delay <configuration_reference_wal_dir_rescan_delay>`
+-   :ref:`wal.max_size <configuration_reference_wal_max_size>`
+-   :ref:`wal.mode <configuration_reference_wal_mode>`
+-   :ref:`wal.queue_max_size <configuration_reference_wal_queue_max_size>`
+-   :ref:`wal.ext.* <configuration_reference_wal_ext>`
+
+    -   :ref:`wal.ext.new <configuration_reference_wal_ext_new>`
+    -   :ref:`wal.ext.old <configuration_reference_wal_ext_old>`
+    -   :ref:`wal.ext.spaces <configuration_reference_wal_ext_spaces>`
+
+..  _configuration_reference_wal_cleanup_delay:
+
+..  confval:: wal.cleanup_delay
+
+    The delay in seconds used to prevent the :ref:`Tarantool garbage collector <configuration_persistence_checkpoint_daemon>`
+    from immediately removing :ref:`write-ahead log <internals-wal>` files after a node restart.
+    This delay eliminates possible erroneous situations when the master deletes WALs
+    needed by :ref:`replicas <replication-roles>` after restart.
+    As a consequence, replicas sync with the master faster after its restart and
+    don't need to download all the data again.
+    Once all the nodes in the replica set are up and running, a scheduled garbage collection is started again
+    even if ``wal_cleanup_delay`` has not expired.
+
+
+    ..  NOTE::
+
+        The option has no effect on nodes running as
+        :ref:`anonymous replicas <configuration_reference_replication_anon>`.
+
+    |
+    | Type: number
+    | Default: 14400
+    | Environment variable: TT_WAL_CLEANUP_DELAY
+
+..  _configuration_reference_wal_dir:
+
+..  confval:: wal.dir
+
+    A directory where write-ahead log (``.xlog`) files are stored.
+    A relative path in this option is interpreted as relative to ``process.work_dir``.
+
+    By default, WAL files and snapshots are stored in the same directory.
+    However, you can set different values for the ``wal.dir`` and :ref:`snapshot.dir <configuration_reference_snapshot_dir>` options
+    to store them on different physical disks for performance matters.
+
+    |
+    | Type: string
+    | Default: 'var/lib/{{ instance_name }}'
+    | Environment variable: TT_WAL_DIR
+
+..  _configuration_reference_wal_dir_rescan_delay:
+
+..  confval:: wal.dir_rescan_delay
+
+    Number of seconds between periodic scans of the write-ahead-log
+    file directory, when checking for changes to write-ahead-log
+    files for the sake of :ref:`replication <replication>` or :ref:`hot standby <index-hot_standby>`.
```

Comment:
Let's remove the link to old config reference (`index-hot_standby`) and add a link to `database.hot_standby` in this reference. It will be documented later: https://github.com/tarantool/doc/issues/3858.